src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42YFlowRttSpec.groovy

```diff
@@ -0,0 +1,672 @@
+            switchHelper.validateAndCollectFoundDiscrepancies(involvedSwitches).isEmpty()
+        }
+
+        and: "Stats for both directions are available for the first sub-flow"
+        Wrappers.wait(STATS_FROM_SERVER42_LOGGING_TIMEOUT, 1) {
+            verifyAll(flowStats.of(yFlow.subFlows.first().flowId)) { subFlow1Stats ->
+                assert subFlow1Stats.get(FLOW_RTT, FORWARD, SERVER_42).hasNonZeroValues()
+                assert subFlow1Stats.get(FLOW_RTT, REVERSE, SERVER_42).hasNonZeroValues()
+            }
+        }
+
+        and: "Stats for both directions are available for the second sub-flow"
+        Wrappers.wait(STATS_FROM_SERVER42_LOGGING_TIMEOUT, 1) {
+            verifyAll(flowStats.of(yFlow.subFlows.last().flowId)) { subFlow2Stats ->
+                assert subFlow2Stats.get(FLOW_RTT, FORWARD, SERVER_42).hasNonZeroValues()
+                assert subFlow2Stats.get(FLOW_RTT, REVERSE, SERVER_42).hasNonZeroValues()
+            }
+        }
+
+        when: "Disable flow rtt on shared switch"
+        //for y-flow shared switch is src sw
+        switchHelper.setServer42FlowRttForSwitch(swT.shared, false)
+        Wrappers.wait(RULES_INSTALLATION_TIME, 3) {
+            assert !switchHelper.validateAndCollectFoundDiscrepancies(swT.shared.dpId).isPresent()
+        }
+
+        then: "Stats are available in REVERSE direction for both sub-flows"
+        def checkpointTime = new Date().getTime() + SERVER42_STATS_LAG * 1000
+
+        Wrappers.wait(STATS_FROM_SERVER42_LOGGING_TIMEOUT, 1) {
+            verifyAll() {
+                assert flowStats.of(yFlow.subFlows.first().flowId).get(FLOW_RTT, REVERSE, SERVER_42).hasNonZeroValuesAfter(checkpointTime)
+                assert flowStats.of(yFlow.subFlows.last().flowId).get(FLOW_RTT, REVERSE, SERVER_42).hasNonZeroValuesAfter(checkpointTime)
+            }
+        }
+
+        and: "Stats are absent in FORWARD direction for both sub-flows"
+        Wrappers.wait(STATS_FROM_SERVER42_LOGGING_TIMEOUT, 1) {
+            verifyAll() {
+                assert !flowStats.of(yFlow.subFlows.first().flowId).get(FLOW_RTT, FORWARD, SERVER_42).hasNonZeroValuesAfter(checkpointTime)
+                assert !flowStats.of(yFlow.subFlows.first().flowId).get(FLOW_RTT, FORWARD, SERVER_42).hasNonZeroValuesAfter(checkpointTime)
+            }
+        }
+
+        cleanup: "Revert system to original state"
+        yFlow && yFlow.delete() && switchHelper.verifyAbsenceOfServer42FlowRttRules(initialSwitchesProps.keySet())
+        flowRttToggleInitialState != null && northbound.toggleFeature(FeatureTogglesDto.builder().server42FlowRtt(flowRttToggleInitialState).build())
+        initialSwitchesProps && switchHelper.revertToOriginSwitchSetup(initialSwitchesProps, flowRttToggleInitialState)
+
+        where:
+        isShEpYPoint << [true, false]
+    }
+
+    def "Y-Flow rtt stats are available only if both global and switch toggles are 'ON' on both endpoints"() {
```

Comment:
This test is 100+ lines long. Please, split it to several smaller tests (or small parametrized test). IIRC, I did something similar in Server42FlowStatsSpec with the same test.

Reply:
Agreed that the test is too long, but it seems like the main concept is to check how the system reacts to the changes in flight. That's why I prefer to leave this implementation as splitting this test will change its logic.